app/cmd/backup.go

```diff
@@ -82,16 +83,53 @@ func getBackupStatus(c *cli.Context, backupID string, replicaAddress string) (*s
 	return backupStatus, nil
 }
 
+func getKnownReplicas(c *cli.Context) (map[string]types.Mode, error) {
+	cli := getCli(c)
+	replicas, err := cli.ReplicaList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get replica list: %v", err)
+	}
+
+	knownReplicas := make(map[string]types.Mode)
+	for _, replica := range replicas {
+		knownReplicas[replica.Address] = replica.Mode
+	}
+
+	return knownReplicas, nil
+}
+
 func fetchAllBackups(c *cli.Context) error {
 	backupProgressList := make(map[string]*sync.BackupStatusInfo)
 
 	cli := getCli(c)
 	backupReplicaMap, err := cli.BackupReplicaMappingGet()
 	if err != nil {
-		return fmt.Errorf("failed to get list of backupIDs:%v", err)
+		return fmt.Errorf("failed to get list of backupIDs: %v", err)
+	}
+
+	knownReplicas, err := getKnownReplicas(c)
+	if err != nil {
+		return err
 	}
 
 	for backupID, replicaAddress := range backupReplicaMap {
+		// a replica is only valid in Read-Write state.
+		// during a rebuild (Write only) the memory information is lost,
+		// so there is no point in requesting the backup status
+		if mode, known := knownReplicas[replicaAddress]; !known || mode != types.RW {
+			err := cli.BackupReplicaMappingDelete(backupID)
```

Comment:
I think we might want to revisit our backup status cleanup policy later. Currently, if a backup was interrupted during the process, the error information with it will be gone as well very soon.

Reply:
Agree but figured it's something for future work.